lib/sshkit/backends/connection_pool.rb

```diff
@@ -103,11 +110,22 @@ def find_cache(args)
   def thread_safe_find_or_create_cache(key)
     caches.synchronize do
       caches[key] ||= begin
-        Cache.new(idle_timeout, method(:silently_close_connection_later))
+        Cache.new(key, idle_timeout, method(:silently_close_connection_later))
       end
     end
   end
 
+  # Update cache key with changed args to prevent cache miss
+  def update_key_if_args_changed(cache, args)
+    new_key = cache_key_for_connection_args(args)
+    return if cache.same_key?(new_key)
+
+    caches.synchronize do
+      cache = caches[new_key] = caches.delete(cache.key)
```

Comment:
Is the chained assignment really needed here? I find it a bit hard to read.

If I remove the extra assignment and just write:

```ruby
caches[new_key] = caches.delete(cache.key)
cache.key = new_key
```

… then the tests still pass.